eventuals/http.h

```diff
@@ -14,27 +15,230 @@ namespace http {
 
 ////////////////////////////////////////////////////////////////////////
 
+enum Method {
+  GET,
+  POST,
+};
+
+////////////////////////////////////////////////////////////////////////
+
 // Used for application/x-www-form-urlencoded case.
 // First string is key, second is value.
 // TODO(folming): switch to RapidJSON.
 using PostFields = std::vector<std::pair<std::string, std::string>>;
 
 ////////////////////////////////////////////////////////////////////////
 
+// Forward declarations.
+template <bool, bool>
+class RequestBuilder;
+
+////////////////////////////////////////////////////////////////////////
+
+class Request {
+ public:
+  // Constructs a new http::Request "builder" with the default
+  // undefined values.
+  static RequestBuilder<false, false> Builder();
+
+  const auto& uri() {
+    return uri_;
+  }
+
+  const auto& method() {
+    return method_;
+  }
+
+  const auto& headers() {
+    return headers_;
+  }
+
+  const auto& body() {
+    return body_;
+  }
+
+  const auto& timeout() {
+    return timeout_;
+  }
+
+  const auto& fields() {
+    return fields_;
+  }
+
+  const auto& verify_peer() {
+    return verify_peer_;
+  }
+
+  const auto& certificate() {
+    return certificate_;
+  }
+
+ private:
+  template <bool, bool>
+  friend class RequestBuilder;
+
+  friend class Client;
+
+  std::string uri_;
+  Method method_;
+  std::vector<std::pair<std::string, std::string>> headers_;
+  std::string body_;
+  std::chrono::nanoseconds timeout_;
+  PostFields fields_;
+  std::optional<bool> verify_peer_;
```

Comment:
We need something like tribool in here: https://www.boost.org/doc/libs/1_78_0/doc/html/boost/logic/tribool.html